src/plugin-hooks/useRowSelect.js

```diff
@@ -167,7 +168,23 @@ function reducer(state, action, previousState, instance) {
       }
     }
 
-    handleRowById(id)
+    if (shiftKey && state.lastSelectedRowId && state.lastSelectedRowId !== id) {
+      const previousIndex = rows.findIndex(
+        row => row.id === state.lastSelectedRowId
+      )
+      const currentIndex = rows.findIndex(row => row.id === id)
+      const startIndex = Math.min(previousIndex, currentIndex)
+      const endIndex = Math.max(previousIndex, currentIndex)
+      const rowIdsToToggle = Object.keys(rowsById).slice(
```

Comment:
don't depend for `rows ids sequence` on `Object.keys(rowsById)`, as it will not change after operations like sort etc. hence selecting rows via shift can produce not required effect.

Reply:
[sidenote] Also, Object.keys does not guarantee order on some envs.

Reply:
Thanks for the feedback. I'll look over it.